taskcluster/ac_taskgraph/transforms/complete.py

```diff
@@ -17,6 +17,11 @@
 @transforms.add
 def fill_dependencies(config, tasks):
     for task in tasks:
+        # TODO: This currently includes some tasks that have been removed from
+        # target tasks. How do we avoid including these here?
+        # Best ideas are:
+        # 1) Move this entire task to a morph (like https://hg.mozilla.org/ci/taskgraph/file/tip/src/taskgraph/morph.py#l187)
```

Comment:
I kinda like this idea, actually 🙂

The complete task as it is now, has this known bug: https://bugzilla.mozilla.org/show_bug.cgi?id=1642977. When I first implemented it, I felt like I was fighting against the Taskcluster paradigm. Moving it to morph would actually solve bug 1642977, if I understand correctly. Moreover, it will be easier to maintain the dependency list of the complete task since we want to wait on all tasks. As of now, we have to manually update `kind-dependencies`.

Reply:
This sounds like good follow-up fodder :). Do you want a new bug or issue for it?